modules/context/repo.go

```diff
@@ -742,12 +744,12 @@ func RepoRefByType(refType RepoRefType) func(*Context) {
 				return
 			}
 			// We opened it, we should close it
-			defer func() {
+			cancel = func() {
 				// If it's been set to nil then assume someone else has closed it.
 				if ctx.Repo.GitRepo != nil {
 					ctx.Repo.GitRepo.Close()
```

Comment:
The gitrepo should be close on the middleware to open it.

Reply:
It is, see Wrap and MiddleCancel below

Reply:
The intention is that the gitrepo is opened once for the entirety of the response.